### ISSUE TITLE
TPL build issues

### ISSUE THREAD

**Issue by samuelpmishLLNL** [LLNL/serac#1019]
When recompiling my TPLs this time on an ubuntu 22.04 machine there have been a handful of issues:

1. The current spack.yaml for ubuntu 22.04 doesn't specify an mpi config, so spack tries to build openmpi automatically and fails:
```
1 error found in build log:
     643    config.status: executing depfiles commands
     644    config.status: executing libtool commands
     645    configure: ===== done with modules/json-c configure =====
     646    checking hwloc.h usability... no
     647    checking hwloc.h presence... no
     648    checking for hwloc.h... no
  >> 649    configure: error: --with-hwloc is given but not found
```
2. Specifying to use mpich triggers spack to try and build it from scratch (despite already being installed by system package manager), and fails with the same error message:
```
1 error found in build log:
     643    config.status: executing depfiles commands
     644    config.status: executing libtool commands
     645    configure: ===== done with modules/json-c configure =====
     646    checking hwloc.h usability... no
     647    checking hwloc.h presence... no
     648    checking for hwloc.h... no
  >> 649    configure: error: --with-hwloc is given but not found
```
Note: hwloc is already installed by the system package manager, and its location / version are specified manually in the ubuntu22/spack.yaml file.

Jamie's helpful suggestion of just ignoring the `ubuntu22/spack.yaml` completely and starting from scratch was able to resolve these two problems, which makes me wonder if those spack environment files are doing more harm than good at this point.

**Comment by samuelpmishLLNL** [comment]
@white238 

building TPLs on a mac doesn't work for me when omitting a spack.yaml file. Running uberenv without it on an M2 mac gives the following error message:

```
==> Error: Spec version is not concrete: autoconf
```

Retrying with a spack.yaml file (the one that worked for this same machine a couple months ago) now fails with

```
==> Error: Spack concretizer internal error. Please submit a bug report and include the command, environment if applicable and the following error message.
    serac@develop%apple-clang is unsatisfiable, errors are:
/Users/mish2/code/serac/scripts/uberenv/uberenv.py:669: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
[ERROR: Failure of spack install]
```

the relevant lines of that python file are:

```py
 666     # Returns version of Spack being used                                        
 667     def spack_version(self):                                                     
 668         res, out = sexe('{0} --version'.format(self.spack_exe(use_spack_env=False)), ret_output=True)
 669         return LooseVersion(out[:-1])   
```

**Comment by chapman39** [comment]
> /Users/mish2/code/serac/scripts/uberenv/uberenv.py:669: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.

I remember seeing this in uberenv. There is a deprecated package in uberenv that should be updated.

**Comment by samuelpmishLLNL** [comment]
after trying some of Chris' suggestions for the mac build, it is still failing with error messages in Lua and Umpire:

```
==> Installing lua-5.4.4-svseejc52stuhxafrzi6hxduxwya2n7s
==> No binary for lua-5.4.4-svseejc52stuhxafrzi6hxduxwya2n7s found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/16/164c7849653b80ae67bec4b7473b884bf5cc8d2dca05653475ec2ed27b9ebf61.tar.gz
==> Fetching https://mirror.spack.io/_source-cache/archive/56/56ab9b90f5acbc42eb7a94cf482e6c058a63e8a1effdf572b8b2a6323a06d923.tar.gz
==> Moving resource stage
        source: /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/resource-luarocks-svseejc52stuhxafrzi6hxduxwya2n7s/spack-src/
        destination: /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-lua-5.4.4-svseejc52stuhxafrzi6hxduxwya2n7s/spack-src/luarocks/luarocks
==> No patches needed for lua
==> lua: Executing phase: 'edit'
==> lua: Executing phase: 'build'
==> lua: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j12' 'build'
See build log for details:
  /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-lua-5.4.4-svseejc52stuhxafrzi6hxduxwya2n7s/spack-build-out.txt
```

and

```
==> Installing umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n
==> No binary for umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n found: installing from source
==> Applied patch /Users/mish2/code/serac/scripts/spack/radiuss-spack-configs/packages/umpire/export_includes.patch
==> umpire: Executing phase: 'initconfig'
==> umpire: Executing phase: 'cmake'
==> umpire: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/usr/bin/make' '-j12'

10 errors found in build log:
     155    cd /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/src/umpire/util && /Users/mish2/code/serac_l
            ibs_11_07_23/spack/lib/spack/env/gcc/g++  -I/Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include -I/var/folders/p_/35z
            ffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src -isystem /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spac
            k-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/tpl -isystem /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2
            gfgs2tadvli7uny5n/spack-build-qeepmm5/include -Wall -Wextra       -Werror  -O3 -DNDEBUG -std=c++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX13.3.sdk -mmacosx-version-min=13.0 -fPIC -fcompare-debug-second -MD -MT src/umpire/util/CMakeFiles/umpire_util.dir/FixedMallocPool.cpp.o -MF CMakeFiles/umpire_util.dir/FixedMallocPool.cpp.o.d -o CM
            akeFiles/umpire_util.dir/FixedMallocPool.cpp.o -c /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire/util
            /FixedMallocPool.cpp
     156    In file included from /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource.hpp:21,
     157                     from /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire/op/MemoryOperation.hpp:12,
     158                     from /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire/op/GenericReallocateOperation.hp
            p:10,
     159                     from /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire/op/GenericReallocateOperation.cp
            p:7:
     160    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource/host.hpp: In member function 'T* camp::resources::v
            1::Host::allocate(size_t, camp::resources::v1::MemoryAccess)':
  >> 161    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource/host.hpp:57:21: error: there are no arguments to 'm
            alloc' that depend on a template parameter, so a declaration of 'malloc' must be available [-fpermissive]
     162       57 |         return (T *)malloc(sizeof(T) * n);
     163          |                     ^~~~~~
     164    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource/host.hpp: In member function 'void camp::resources:
            :v1::Host::deallocate(void*, camp::resources::v1::MemoryAccess)':
  >> 165    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource/host.hpp:65:71: error: 'free' was not declared in t
            his scope
     166       65 |       void deallocate(void *p, MemoryAccess = MemoryAccess::Device) { free(p); }
     167          |                                                                       ^~~~
     168    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource.hpp: In member function 'T camp::resources::v1::Res
            ource::get()':
  >> 169    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource.hpp:72:22: error: 'runtime_error' is not a member o
            f 'std'
     170       72 |           throw std::runtime_error("Incompatible Resource type get cast.");
     171          |                      ^~~~~~~~~~~~~
     172    [ 31%] Building CXX object src/umpire/event/CMakeFiles/umpire_event.dir/json_file_store.cpp.o
     173    cd /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/src/umpire/event && /Users/mish2/code/serac_
            libs_11_07_23/spack/lib/spack/env/gcc/g++  -I/Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include -I/var/folders/p_/35
            zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src -isystem /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spa
            ck-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/tpl -isystem /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s
            2gfgs2tadvli7uny5n/spack-build-qeepmm5/include -Wall -Wextra       -Werror  -O3 -DNDEBUG -std=c++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDK
            s/MacOSX13.3.sdk -mmacosx-version-min=13.0 -fPIC -fcompare-debug-second -MD -MT src/umpire/event/CMakeFiles/umpire_event.dir/json_file_store.cpp.o -MF CMakeFiles/umpire_event.dir/json_file_store.cpp.o.d -
            o CMakeFiles/umpire_event.dir/json_file_store.cpp.o -c /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire
            /event/json_file_store.cpp
     174    [ 33%] Building CXX object src/umpire/event/CMakeFiles/umpire_event.dir/recorder_factory.cpp.o
     175    cd /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/src/umpire/event && /Users/mish2/code/serac_
            libs_11_07_23/spack/lib/spack/env/gcc/g++  -I/Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include -I/var/folders/p_/35
            zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src -isystem /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spa
            ck-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/tpl -isystem /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s
            2gfgs2tadvli7uny5n/spack-build-qeepmm5/include -Wall -Wextra       -Werror  -O3 -DNDEBUG -std=c++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDK
            s/MacOSX13.3.sdk -mmacosx-version-min=13.0 -fPIC -fcompare-debug-second -MD -MT src/umpire/event/CMakeFiles/umpire_event.dir/recorder_factory.cpp.o -MF CMakeFiles/umpire_event.dir/recorder_factory.cpp.o.d
             -o CMakeFiles/umpire_event.dir/recorder_factory.cpp.o -c /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/ump
            ire/event/recorder_factory.cpp
     176    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource/host.hpp: In instantiation of 'T* camp::resources::
            v1::Host::allocate(size_t, camp::resources::v1::MemoryAccess) [with T = char; size_t = long unsigned int]':
     177    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource/host.hpp:61:33:   required from here
  >> 178    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource/host.hpp:57:27: error: 'malloc' was not declared in
             this scope, and no declarations were found by argument-dependent lookup at the point of instantiation
     179       57 |         return (T *)malloc(sizeof(T) * n);
     180          |                     ~~~~~~^~~~~~~~~~~~~~~
  >> 181    make[2]: *** [src/umpire/op/CMakeFiles/umpire_op.dir/GenericReallocateOperation.cpp.o] Error 1
  >> 182    make[1]: *** [src/umpire/op/CMakeFiles/umpire_op.dir/all] Error 2
     183    make[1]: *** Waiting for unfinished jobs....
     184    [ 35%] Building CXX object src/umpire/strategy/CMakeFiles/umpire_strategy.dir/AllocationAdvisor.cpp.o
     185    cd /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/src/umpire/strategy && /Users/mish2/code/ser
            ac_libs_11_07_23/spack/lib/spack/env/gcc/g++  -I/Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include -I/var/folders/p_
            /35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src -isystem /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/
            spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/tpl -isystem /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4j
            z2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/include -Wall -Wextra       -Werror  -O3 -DNDEBUG -std=c++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/
            SDKs/MacOSX13.3.sdk -mmacosx-version-min=13.0 -fPIC -fcompare-debug-second -MD -MT src/umpire/strategy/CMakeFiles/umpire_strategy.dir/AllocationAdvisor.cpp.o -MF CMakeFiles/umpire_strategy.dir/AllocationA
            dvisor.cpp.o.d -o CMakeFiles/umpire_strategy.dir/AllocationAdvisor.cpp.o -c /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n
            /spack-src/src/umpire/strategy/AllocationAdvisor.cpp
     186    [ 37%] Building CXX object src/umpire/util/CMakeFiles/umpire_util.dir/io.cpp.o
     187    cd /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/src/umpire/util && /Users/mish2/code/serac_l
            ibs_11_07_23/spack/lib/spack/env/gcc/g++  -I/Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include -I/var/folders/p_/35z
            ffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src -isystem /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spac
            k-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/tpl -isystem /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2
            gfgs2tadvli7uny5n/spack-build-qeepmm5/include -Wall -Wextra       -Werror  -O3 -DNDEBUG -std=c++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX13.3.sdk -mmacosx-version-min=13.0 -fPIC -fcompare-debug-second -MD -MT src/umpire/util/CMakeFiles/umpire_util.dir/io.cpp.o -MF CMakeFiles/umpire_util.dir/io.cpp.o.d -o CMakeFiles/umpire_util.dir/i
            o.cpp.o -c /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire/util/io.cpp
     188    [ 39%] Building CXX object src/umpire/event/CMakeFiles/umpire_event.dir/quest_database.cpp.o

     ...

     209    cd /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/src/umpire/strategy && /Users/mish2/code/ser
            ac_libs_11_07_23/spack/lib/spack/env/gcc/g++  -I/Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include -I/var/folders/p_
            /35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src -isystem /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/
            spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/tpl -isystem /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4j
            z2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/include -Wall -Wextra       -Werror  -O3 -DNDEBUG -std=c++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/
            SDKs/MacOSX13.3.sdk -mmacosx-version-min=13.0 -fPIC -fcompare-debug-second -MD -MT src/umpire/strategy/CMakeFiles/umpire_strategy.dir/mixins/AllocateNull.cpp.o -MF CMakeFiles/umpire_strategy.dir/mixins/Al
            locateNull.cpp.o.d -o CMakeFiles/umpire_strategy.dir/mixins/AllocateNull.cpp.o -c /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli
            7uny5n/spack-src/src/umpire/strategy/mixins/AllocateNull.cpp
     210    [ 62%] Building CXX object src/umpire/util/CMakeFiles/umpire_util.dir/OutputBuffer.cpp.o
     211    cd /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/src/umpire/util && /Users/mish2/code/serac_l
            ibs_11_07_23/spack/lib/spack/env/gcc/g++  -I/Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include -I/var/folders/p_/35z
            ffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src -isystem /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spac
            k-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/tpl -isystem /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2
            gfgs2tadvli7uny5n/spack-build-qeepmm5/include -Wall -Wextra       -Werror  -O3 -DNDEBUG -std=c++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX13.3.sdk -mmacosx-version-min=13.0 -fPIC -fcompare-debug-second -MD -MT src/umpire/util/CMakeFiles/umpire_util.dir/OutputBuffer.cpp.o -MF CMakeFiles/umpire_util.dir/OutputBuffer.cpp.o.d -o CMakeFil
            es/umpire_util.dir/OutputBuffer.cpp.o -c /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire/util/OutputBu
            ffer.cpp
     212    In file included from /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire/ResourceManager.hpp:17,
     213                     from /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire/strategy/mixins/AllocateNull.cpp
            :9:
     214    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource.hpp: In member function 'T camp::resources::v1::Res
            ource::get()':
  >> 215    /Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include/camp/resource.hpp:72:22: error: 'runtime_error' is not a member o
            f 'std'
     216       72 |           throw std::runtime_error("Incompatible Resource type get cast.");
     217          |                      ^~~~~~~~~~~~~
  >> 218    make[2]: *** [src/umpire/strategy/CMakeFiles/umpire_strategy.dir/mixins/AllocateNull.cpp.o] Error 1
     219    make[2]: *** Waiting for unfinished jobs....
     220    [ 64%] Building CXX object src/umpire/util/CMakeFiles/umpire_util.dir/allocation_statistics.cpp.o
     221    cd /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/src/umpire/util && /Users/mish2/code/serac_l
            ibs_11_07_23/spack/lib/spack/env/gcc/g++  -I/Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include -I/var/folders/p_/35z
            ffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src -isystem /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spac
            k-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/tpl -isystem /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2
            gfgs2tadvli7uny5n/spack-build-qeepmm5/include -Wall -Wextra       -Werror  -O3 -DNDEBUG -std=c++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX13.3.sdk -mmacosx-version-min=13.0 -fPIC -fcompare-debug-second -MD -MT src/umpire/util/CMakeFiles/umpire_util.dir/allocation_statistics.cpp.o -MF CMakeFiles/umpire_util.dir/allocation_statistics.c
            pp.o.d -o CMakeFiles/umpire_util.dir/allocation_statistics.cpp.o -c /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-s
            rc/src/umpire/util/allocation_statistics.cpp
     222    [ 66%] Building CXX object src/umpire/util/CMakeFiles/umpire_util.dir/detect_vendor.cpp.o
     223    cd /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-build-qeepmm5/src/umpire/util && /Users/mish2/code/serac_l
            ibs_11_07_23/spack/lib/spack/env/gcc/g++  -I/Users/mish2/code/serac_libs_11_07_23/spack/opt/spack/darwin-ventura-m2/gcc-13.1.0/camp-2022.10.1-xrhjjdr22caunels6m3zbzn4nv6zgsyh/include -I/var/folders/p_/35z
            ffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src -isystem /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spac
            k-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/tpl -isystem /private/var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2
            gfgs2tadvli7uny5n/spack-build-qeepmm5/include -Wall -Wextra       -Werror  -O3 -DNDEBUG -std=c++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX13.3.sdk -mmacosx-version-min=13.0 -fPIC -fcompare-debug-second -MD -MT src/umpire/util/CMakeFiles/umpire_util.dir/detect_vendor.cpp.o -MF CMakeFiles/umpire_util.dir/detect_vendor.cpp.o.d -o CMakeF
            iles/umpire_util.dir/detect_vendor.cpp.o -c /var/folders/p_/35zffh5d0ns1zpr0ddx1fn4r000trx/T/mish2/spack-stage/spack-stage-umpire-2022.10.0-qeepmm5bm4jz2s2gfgs2tadvli7uny5n/spack-src/src/umpire/util/detec
            t_vendor.cpp
     224    [ 66%] Built target umpire_resource
     225    [ 66%] Built target umpire_interface
     226    [ 66%] Built target umpire_util
  >> 227    make[1]: *** [src/umpire/strategy/CMakeFiles/umpire_strategy.dir/all] Error 2
     228    [ 66%] Built target umpire_event
  >> 229    make: *** [all] Error 2
```

**Comment by samuelpmishLLNL** [comment]
for my records, I'm hitting the umpire issues in a recent build too. 

At the time of writing this, our version of umpire accidentally put some gcc-specific flags (-fcompare-debug-second) into their cmake targets, which makes everything downstream of that unable to build with clang. The fix was to manually open up the spack installation directory for umpire and patch the `umpire-targets.cmake` file to omit those flags.

**Comment by white238** [comment]
> for my records, I'm hitting the umpire issues in a recent build too.
> 
> At the time of writing this, our version of umpire accidentally put some gcc-specific flags (-fcompare-debug-second) into their cmake targets, which makes everything downstream of that unable to build with clang. The fix was to manually open up the spack installation directory for umpire and patch the `umpire-targets.cmake` file to omit those flags.

This will be addressed in #1052. Umpire removed that flag in October 2023 and my branch moves us to the 2024.02.0 release of umpire.

I'm close I promise!
